pkg/operator/ceph/object/rgw.go

```diff
@@ -145,6 +146,15 @@ func (c *clusterConfig) startRGWPods(realmName, zoneGroupName, zoneName string)
 			}
 		}
 
+		rgwsToSkipReconcile, err := mon.ToSkipReconcile(c.clusterInfo.Context, c.context, c.clusterInfo.Namespace, config.RgwType, AppName)
```

Comment:
This query could move up before line 109 so it's outside the `for` loop so it's only run once.